loki/visitors/pretty_visualize.py

```diff
@@ -0,0 +1,349 @@
+        kwargs["parent"]: :any: `Node` object, optional (default: None)
+            If not available no edge is drawn.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            A list of a tuple of a node and potentially a edge information
+        """
+        label = kwargs.get("label", "")
+        if label == "":
+            label = self.format_node(repr(node))
+
+        shape = kwargs.get("shape", "oval")
+        label = (
+            label + "‎"
+        )  # dirty hack to force graphviz to utilize parenthesis around the label
+
+        node_key = str(id(node))
+        if node_key not in self._id_map:
+            self._id_map[node_key] = str(self._id)
+            self._id += 1
+
+        node_info = {
+            "name": str(self._id_map[node_key]),
+            "label": str(label),
+            "shape": str(shape),
+        }
+
+        parent = kwargs.get("parent")
+        edge_info = {}
+        if parent:
+            parent_id = self._id_map[str(id(parent))]
+            child_id = self._id_map[str(id(node))]
+            edge_info = {"tail_name": str(parent_id), "head_name": str(child_id)}
+
+        return [(node_info, edge_info)]
+
+    # Handler for outer objects
+    def visit_Module(self, o, **kwargs):
+        """
+        Add a :any:`Module`, mark parent node and visit all "spec" and "subroutine" nodes.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information
+        """
+        node_edge_info = self.__add_node(o, **kwargs)
+        kwargs["parent"] = o
+
+        node_edge_info.extend(self.visit(o.spec, **kwargs))
+        node_edge_info.extend(self.visit(o.subroutines, **kwargs))
+
+        return node_edge_info
+
+    def visit_Subroutine(self, o, **kwargs):
+        """
+        Add a :any:`Subroutine`, mark parent node and visit all "docstring", "spec", "body", "members" nodes.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information
+        """
+        node_edge_info = self.__add_node(o, **kwargs)
+        kwargs["parent"] = o
+
+        node_edge_info.extend(self.visit(o.docstring, **kwargs))
+        node_edge_info.extend(self.visit(o.spec, **kwargs))
+        node_edge_info.extend(self.visit(o.body, **kwargs))
+        node_edge_info.extend(self.visit(o.members, **kwargs))
+
+        return node_edge_info
+
+    # Handler for AST base nodes
+    def visit_Comment(self, o, **kwargs):
+        """
+        Enables turning off comments.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information, or list of nothing.
+        """
+        if kwargs.get("show_comments"):
```

Comment:
This is behaviour parameterisation for the entire class - hence I would suggest to make this an object attribute and to add it to the constructor instead of passing it around via `kwargs`.

Reply:
this and show comments are now object attributes passed to the constructor c72fe87359edd81a3f233e85478edf849f7b2c33